HIP/hip-1272.md

```diff
@@ -0,0 +1,78 @@
+---
+hip: 1272
+title: Unified Mirror Node entity endpoint for shared entity metadata
+author: Michael Kantor (@kantorcodes)
+requested-by: Hashgraph Online
+type: Standards Track
+category: Mirror
+status: Draft
+discussions-to:
+created: 2025-08-24
+updated: 2025-08-24
+---
+
+## Abstract:
+
+Introduce a new Mirror Node API endpoint, `GET /api/v1/entities/{id}`, that resolves any Hedera entity identifier (shard.realm.num or alias) and returns a compact, normalized view with the entity type and a minimal set of common, cross-entity fields. This reduces ambiguity when a client only has an identifier and needs to determine the entity type and basic metadata without calling multiple type-specific endpoints.
+
+## Motivation:
+
+Today, clients must guess an entity's type and probe multiple endpoints (e.g., `/accounts/{id}`, `/contracts/{id}`, `/tokens/{id}`, `/topics/{id}`, etc.). This is cumbersome when:
+
+- A user pastes an identifier of unknown type
+- The identifier is an alias that could map to an account
+- A workflow needs only shared metadata (creation time, deletion status, ledger, memo) before navigating to type-specific details
+
+A single resolver endpoint improves developer experience, reduces round trips, and establishes a stable contract for cross-entity discovery that aligns with how Mirror Node already models entities internally. Including this endpoint should reduce egress traffic and improve performance.
+
+## Rationale:
+
+- A unified resolver reflects the Mirror Node's normalized `entity` model.
+- Returning a minimal, common shape keeps the response stable over time and avoids leaking type-specific concerns.
+- Supporting multiple identifier forms (0.0.x, alias) matches existing Mirror Node resolution behavior and avoids fragmentation.
+
+## User stories
+
+- As a wallet, I want to paste any identifier and immediately know if it is an account, contract, token, topic, file, or schedule so I can route users appropriately.
+- As an indexer, I want a single API to normalize identifiers to `0.0.x` and detect the entity type before fetching type-specific details.
+- As a developer, I want to quickly check basic entity metadata (created timestamp, deletion status, memo) without guessing which endpoint to query.
+
+## Specification:
+
+The REST API will be extended with a unified resolver endpoint that accepts any Hedera entity identifier and returns shared entity metadata.
+
+### Route:
+
+/api/v1/entities/{id}
+
+### Proposed Response Structure:
+
+```json
+{
+  "entity_id": "0.0.1234",
+  "type": "ACCOUNT|CONTRACT|TOKEN|TOPIC|FILE|SCHEDULE|NODE",
```

Comment:
Today the Mirror Node does not have endpoints for all the mentioned types (like File). Should we support all directly or concentrate on the types that are supported today?